### PR TITLE
Add support for cracking AxCrypt 2.x

### PR DESCRIPTION
**Pull request by kholia**
This fixes issue #3354 (Add support for cracking AxCrypt 2.x ).

### CODE REVIEW

File: run/axcrypt2john.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/python
 
-# AxCrypt 1.x encrypted file parser for JtR.
+# AxCrypt 1.x and 2.x encrypted file parser for JtR.
+#
 # Written in 2016 by Fist0urs <eddy.maaalou at gmail.com>.
 #
 # This software is Copyright (c) 2016, Fist0urs <eddy.maaalou at gmail.com>,
```
Comment (magnumripper) [inline run/axcrypt2john.py]: You should probably add your own copyright/license blurb as well here

Comment (magnumripper) [reply run/axcrypt2john.py]: Can't fool the bots though
```
opencl_axcrypt_fmt_plug.c:287:3: error: ‘axcrypt_tests’ undeclared here (not in a function)
   axcrypt_tests
   ^
make[1]: *** [opencl_axcrypt_fmt_plug.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [default] Error 2
```
Just a super quick guess is you forgot to source the variable header

Comment (kholia) [reply run/axcrypt2john.py]: Yep, I wrote this on a computer without a working OpenCL setup. Fixing now.

Comment (kholia) [reply run/axcrypt2john.py]: License stuff should be fixed too now.

File: src/axcrypt_fmt_plug.c [hunk 2]
```diff
@@ -59,6 +69,12 @@ john_register_one(&fmt_axcrypt);
 		(cp)[2] ^= (unsigned char)((value) >> 16), \
 		(cp)[3] ^= (unsigned char)((value) >> 24 ) )
 
+#define PUT_64BITS_XOR_LSB(cp, value) ( \
```
Comment (kholia) [inline src/axcrypt_fmt_plug.c]: I have no idea what to call this stuff - I just made this name up ;)

**Comment by kholia** [comment]
I found https://github.com/0xd4d/dnSpy quite useful for debugging this stuff. 

I was able to build https://bitbucket.org/axantum/axcrypt-net in Visual Studio 2017 Community Edition running under Windows 7 64-bit.

**Comment by magnumripper** [comment]
I'd rather eat a discipline ball a whole weekend.

**Pull request merged**